### ISSUE TITLE
Update server database to store `StorageKey`s

### ISSUE THREAD

**Issue by marsella** [boltlabs-inc/key-mgmt#124]
In the [registration protocol](https://github.com/boltlabs-inc/key-mgmt-spec/blob/main/system-functionalities.md#register), the server needs to store a validated, encrypted storage key in a record associated with the user. 

The storage key should only be set once (during registration), and will be retrieved often.

- [ ] Update the [`dams::user::User`] model to include an encrypted storage key (blocked on #113 - this type isn't defined yet)
- [ ] Implement adding an encrypted storage key to a user in  `dams-key-server::database::database`
- [ ] Implement retrieving an encrypted storage key from a user in `dams-key-server::database::database`
- [ ] Test that storing and retrieving a key gets you the same key back.

**Comment by picklenerd** [comment]
On hold while crypto gets finished
